x-pack/plugins/actions/server/routes/create.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import {
+  IRouter,
+  RequestHandlerContext,
+  KibanaRequest,
+  IKibanaResponse,
+  KibanaResponseFactory,
+} from 'kibana/server';
+import { ActionResult } from '../types';
+import { LicenseState } from '../lib/license_state';
+import { verifyApiAccess } from '../lib/license_api_access';
+
+export const bodySchema = schema.object({
+  name: schema.string(),
+  actionTypeId: schema.string(),
+  config: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
+  secrets: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
+});
+
+export const createActionRoute = (router: IRouter, licenseState: LicenseState) => {
+  router.post(
+    {
+      path: `/api/action`,
+      validate: {
+        body: bodySchema,
+      },
+      options: {
+        tags: ['access:actions-all'],
+      },
+    },
+    router.handleLegacyErrors(async function(
+      context: RequestHandlerContext,
+      req: KibanaRequest<any, any, TypeOf<typeof bodySchema>, any>,
+      res: KibanaResponseFactory
+    ): Promise<IKibanaResponse<any>> {
+      verifyApiAccess(licenseState);
```

Comment:
I guess we used some "middleware" to do this before, that capability isn't available in NP?  Or maybe the state we needed wasn't available where we used to need it?  Unfortunate.

Reply:
There doesn't seem to be this equivalent middleware in NP.
The state is the same as before, we did this verify manually at setup, now its done manually at the call site.

It's not ideal, but at least its more visible to maintainers now.